shop/src/main/java/ro/msg/learning/shop/services/OrderService.java

```diff
@@ -0,0 +1,105 @@
+package ro.msg.learning.shop.services;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.configurations.StrategyConfiguration;
+import ro.msg.learning.shop.dtos.OrderDTO;
+import ro.msg.learning.shop.dtos.OrderDetailDTO;
+import ro.msg.learning.shop.dtos.ProductDTO;
+import ro.msg.learning.shop.dtos.StockDTO;
+import ro.msg.learning.shop.entities.*;
+import ro.msg.learning.shop.exceptions.ProductNotFoundException;
+import ro.msg.learning.shop.exceptions.ProductsCantBeShipped;
+import ro.msg.learning.shop.mappers.OrderDetailMapper;
+import ro.msg.learning.shop.mappers.OrderMapper;
+import ro.msg.learning.shop.mappers.ProductMapper;
+import ro.msg.learning.shop.repositories.*;
+import ro.msg.learning.shop.utils.LocationFormat;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+public class OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderDetailRepository orderDetailsRepository;
+    private final OrderDetailMapper orderDetailMapper;
+    private final ProductRepository productRepository;
+    private final ProductMapper productMapper;
+    private final CustomerRepository customerRepository;
+    private final StrategyConfiguration strategyConfiguration;
+    private final LocationRepository locationRepository;
+
+
+    public OrderDTO createOrder(OrderDTO orderDTO) {
+        OrderDTO newOrder;
+        LocalDateTime localDateTime = LocalDateTime.now();
+        List<OrderDetail> result = registerOrderedProducts(orderDTO.getOrderedProducts(), false);
+        LocationFormat deliveryLocation = new LocationFormat(null, orderDTO.getAddress().getCity(), orderDTO.getAddress().getCountry());
+        List<StockDTO> strategyResult = strategyConfiguration.decideStrategy().implementStrategy(orderDetailMapper.orderDetailListToOrderDetailDTOList(result), deliveryLocation);
+        Set<Location> locationsForShippingProducts = getShippingLocations(strategyResult);
+        if (!strategyResult.isEmpty()) {
+            Optional<Customer> client = customerRepository.findById(orderDTO.getUserID());
```

Comment:
this method not only registers the order, but also looks for the customer. It could be splitted up into more methods to better reflect their purpose. Why? -> Separation of concerns.

The best rule of thumb to keep in mind: Controllers communicate with Services. Services communicate with their designated repo (this case only with OrderRepository) and with other services. Refactor this class in order to reflect this arhitectural design.